Fix/834 kakao ridirect url

jye326 opened the pull request:
fix: 카카오 OAuth 로그인 redirectUrl 제거 및 동적 처리 개선

- `redirectUrl` 파라미터 제거 및 `application.yml` 설정값으로 대체
- `OauthClientService`에서 `redirectUrl` 필드 추가 및 관련 로직 수정
- `kakaoCallback` 메서드 HTTP 메서드 POST로 재변경
- 테스트 및 어플리케이션 설정 파일 수정 (application.yml, application-test.yml)

jye326 committed:
fix: 카카오 OAuth 로그인 redirectUrl 제거 및 동적 처리 개선

- `redirectUrl` 파라미터 제거 및 `application.yml` 설정값으로 대체
- `OauthClientService`에서 `redirectUrl` 필드 추가 및 관련 로직 수정
- `kakaoCallback` 메서드 HTTP 메서드 POST로 재변경
- 테스트 및 어플리케이션 설정 파일 수정 (application.yml, application-test.yml)

jye326 committed:
feat: AuthService에 @Transactional 추가 및 OAuth 관련 메서드 리팩토링

- `kakaoLogin` 메서드에 `@Transactional` 어노테이션 추가
- `registerOauthMember` 메서드에 `@Transactional` 어노테이션 추가
- 코드의 트랜잭션 관리 개선 및 안정성 강화

ppparkta commented on backend/fittoring/src/main/java/fittoring/mentoring/business/service/AuthService.java:
```diff
@@ -113,8 +113,9 @@ public void logout(Long memberId) {
 
-    public AuthTokenResponse kakaoLogin(String code, String redirectUrl) {
-        KakaoTokenResponse tokenResponse = oauthClientService.requestKakaoToken(code, redirectUrl);
+    @Transactional
```
p1: 이 서비스에 트랜잭션은 필요하지 않아 보이네요!

ppparkta commented on backend/fittoring/src/main/java/fittoring/mentoring/infra/OauthClientService.java:
```diff
@@ -16,14 +16,16 @@ public class OauthClientService {
     private final RestClient restClient;
     private static final String kakaoTokenRequestUri = "https://kauth.kakao.com/oauth/token";
     private static final String kakaoUserInfoRequestUri = "https://kapi.kakao.com/v2/user/me";
+    @Value("${kakao.redirect-url}")
+    private String redirectUrl;
```
p3: redirect-url을 서버에서 갖고 가게 수정되었네요. 좋습니다!

ppparkta reviewed (approved):
트랜잭션만 제거되면 바로 머지돼도 좋을 것 같아요! 고생하셨습니다.

jye326 committed:
fix: Kakao OAuth 설정값 수정 및 @Transactional 어노테이션 제거

- `KAKAO_REDIRECT_URI`를 `KAKAO_REDIRECT_URL`로 수정
- 워크플로우 설정 파일의 카카오 로그인 콜백 URL 값 변경
- `AuthService`의 `kakaoLogin` 메서드에서 `@Transactional` 제거
